File: Graphs/DirectedGraphs/examples/directed_graph_example.js
const DirectedGraph = require('../source/directed_graph');

// Create a directed graph
const graph = new DirectedGraph();

// Add vertices to the graph
graph.addVertex('A');
graph.addVertex('B');
graph.addVertex('C');
graph.addVertex('D');

// Add directed edges between vertices
graph.addEdge('A', 'B'); // A -> B
graph.addEdge('A', 'C'); // A -> C
graph.addEdge('B', 'D'); // B -> D
graph.addEdge('C', 'D'); // C -> D

// Print the graph's adjacency list
console.log('Graph Adjacency List:');
graph.print();

// Check if specific edges exist
console.log('\nChecking if edges exist:');
console.log(`Does an edge exist from A to B? ${graph.hasEdge('A', 'B') ? 'Yes' : 'No'}`);
console.log(`Does an edge exist from A to D? ${graph.hasEdge('A', 'D') ? 'Yes' : 'No'}`);
console.log(`Does an edge exist from B to D? ${graph.hasEdge('B', 'D') ? 'Yes' : 'No'}`);

File: Graphs/DirectedGraphs/examples/directed_graph_example.py
from Graphs.DirectedGraphs.source.directed_graph import DirectedGraph

if __name__ == "__main__":
    # Create a directed graph
    graph = DirectedGraph()

    # Add vertices to the graph
    graph.add_vertex('A')
    graph.add_vertex('B')
    graph.add_vertex('C')
    graph.add_vertex('D')

    # Add directed edges between vertices
    graph.add_edge('A', 'B')  # A -> B
    graph.add_edge('A', 'C')  # A -> C
    graph.add_edge('B', 'D')  # B -> D
    graph.add_edge('C', 'D')  # C -> D

    # Print the graph's adjacency list
    print("Graph Adjacency List:")
    graph.print()

    # Check if specific edges exist
    print("\nChecking if edges exist:")
    print(f"Does an edge exist from A to B? {'Yes' if graph.has_edge('A', 'B') else 'No'}")
    print(f"Does an edge exist from A to D? {'Yes' if graph.has_edge('A', 'D') else 'No'}")
    print(f"Does an edge exist from B to D? {'Yes' if graph.has_edge('B', 'D') else 'No'}")
    print(f"Does an edge exist from C to D? {'Yes' if graph.has_edge('C', 'D') else 'No'}")
